safe/gui/tools/wizard/wizard_utils.py

```diff
@@ -192,26 +193,42 @@ def get_inasafe_default_value_fields(qsetting, field_key):
 def clear_layout(layout):
     """Clear layout content.
 
-    # Adapted from http://stackoverflow.com/a/9375273/1198772
+    # Adapted from http://stackoverflow.com/a/9383780
+    # and http://stackoverflow.com/a/9375273/1198772
 
     :param layout: A layout.
     :type layout: QLayout
     """
-    for i in reversed(range(layout.count())):
-        item = layout.itemAt(i)
-
-        if isinstance(item, QWidgetItem):
-            item.widget().close()
-        elif isinstance(item, QLayout):
-            clear_layout(item.layout())
-        elif isinstance(item, QSpacerItem):
-            # No need to do anything
-            pass
-        else:
-            pass
+    # Different platform has different treatment
+    # If InaSAFE running on Windows or Linux
+    if 'linux2' or 'win32' == sys.platform:
+        if layout is not None:
+            while layout.count():
+                item = layout.takeAt(0)
+                widget = item.widget()
+                if widget is not None:
+                    # Remove the item from layout
+                    widget.deleteLater()
+                else:
+                    clear_layout(item.layout())
+
+    # If InaSAFE running on Mac OS
+    elif 'darwin' == sys.platform:
```

Comment:
Just to be sure, maybe change this one to a 'else' in case if some people come with a FreeBSD ! :D